pkg/controller/windowsmachine/windows/windows.go

```diff
@@ -288,6 +305,32 @@ func (vm *windows) ConfigureKubeProxy(nodeName, hostSubnet string) error {
 
 // Interface helper methods
 
+// isHostNameChangeNeeded tells if we need to update the host name of the Windows VM
```

Comment:
You are making a subtle assumption that the hostname of a VM corresponds to Machine.Name. I would rather that we not go down that route. I understand you want to be provider agnostic but now you are running the risk of doing something not needed for non-vSphere cloud providers that could result in unknown behavior. I prefer being provider specific in WMCO and remove that when can make MAO handle this.